src/Utilities/Requires.hpp

```diff
@@ -0,0 +1,44 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+/// \file
+/// Defines the type alias Requires
+
+#include <cstddef>
+
+namespace Requires_detail {
+template <bool B>
+struct requires_impl {
+  using Requires_not_satisfied_see_functions_or_classes_Requires_parameters =
+      std::nullptr_t;
+};
+
+template <>
+struct requires_impl<false> {};
+}  // namespace Requires_detail
+
+/*!
+ * \ingroup TypeTraits
+ * \ingroup Utilities
+ * \brief Express requirements on the template parameters of a function or
+ * class, replaces `std::enable_if_t`
+ *
+ * Replacement for `std::enable_if_t` and Concepts for expressing requirements
+ * on template parameters. However, it does not require merging of the Concepts
+ * TS (whose merit is debatable) and provides an "error message" if substitution
+ * of a template parameter failed. Specifically, the compiler error will contain
+ * "Requires_not_satisfied_see_functions_requires_parameters", aiding the user
```

Comment:
Here I'm happy to go over the limit of 80 characters since it will actually improve code usability significantly, Steph and I came up with "template_error_because_types_failed_to_meet_requirements_on_template_parameters"